### ISSUE TITLE
Does this script work to convert RedHat 8 to Oracle Linux 8 ?

### ISSUE THREAD

**Issue by ehb54** [oracle/centos2ol#125]
i have had good success converting several CentOS 8 systems to Oracle Linux.
Now I have some systems with RedHat 8 that also need conversion.
Has anyone tried with this script?

**Comment by Djelibeybi** [comment]
It is not designed for that, but it might work, if you remove all the `rhsm` related packages first.

The official switch process is documented here: https://linux.oracle.com/switch.html

**Comment by ehb54** [comment]
Thank you Djelibeybi for the proper link.

**Comment by ehb54** [comment]
Thanks @Djelibeybi.
The official switch process seems to refer a RH subscription->OL subscription, so it wasn't what we wanted. After converting 2 systems from RH to OL8 leveraging this script, I can say it is possible to do. Someone with greater expertise could probably come up with a simpler method.

My observations:

The `centos2ol.sh` script will fail but seemed good to get started.
After it fails, I had to run
```
wget http://public-yum.oracle.com/RPM-GPG-KEY-oracle-ol8 -O /etc/pki/rpm-gpg/RPM-GPG-KEY-oracle
dnf remove libreport
dnf update
yum distro-sync
```
Not sure if I needed to do the `dnf update` before `yum distro-sync` or that `dnf update` could be skipped. 
Then I created a modified version of the `centos2ol.sh` script which just included the steps after the `yum distro-sync` & ran that (again, not sure if this was needed).

Then duplicated the contents of `/etc/yum.repos.d` from a working OL8 system.
```
echo > /etc//yum/vars/ociregion
echo oracle.com > /etc/yum/vars/ocidomain
dnf update
rpm --erase $( rpm -qa | egrep 'subscription-manager|rhn|redhat-support|redhat-access' )
rm /etc/motd.d/insights-client
dnf install kernel-uek
dnf update (repeat until “nothing to do”)
reboot
```

This is what worked for me... but I'm not recommending it.
This has only been tested on 2 systems with identical original configurations.
I also had to ( as in #8 )
```
dnf module reset llvm-toolset:ol8
dnf module install llvm-toolset:ol8
dnf module reset virt
dnf module install virt
```
